pkg/api/datasources.go

```diff
@@ -316,6 +317,10 @@ func (hs *HTTPServer) CallDatasourceResource(c *models.ReqContext) {
 }
 
 func convertModelToDtos(ds *models.DataSource) dtos.DataSource {
+	if ds.Type == "stackdriver" { // legacy
```

Comment:
Code smell, this logic should reside in plugins package. See my comment on plugin.json